server/provider/scheduledMiniblocks.go

```diff
@@ -0,0 +1,92 @@
+package provider
+
+import (
+	"github.com/ElrondNetwork/elrond-go-core/data/transaction"
+	"github.com/ElrondNetwork/elrond-proxy-go/data"
+)
+
+func (provider *networkProvider) simplifyBlockWithScheduledTransactions(block *data.Block) error {
+	// Simply ignore miniblocks with processing type "processed"
+	filterOutProcessedMiniblocksOnceScheduled(block)
+
+	// Let's check if there's any scheduled miniblock "from me" in the block.
+	// If so, special handling of invalid transactions is required.
+	containsScheduledMiniblocksFromMe := provider.blockContainsScheduledMiniblockFromMe(block)
+	if !containsScheduledMiniblocksFromMe {
+		return nil
+	}
+
+	// If so, get the block N+1 and look for invalid miniblocks holding transactions that were scheduled in block N.
+	// Strip them from block N (where scheduled).
+	nextBlock, err := provider.GetBlockByNonce(block.Nonce + 1)
+	if err != nil {
+		return err
+	}
+
+	invalidTxsInNextBlock := findInvalidTransactionsHashes(nextBlock)
+	filterOutTransactionsFromScheduledMiniblocks(block, invalidTxsInNextBlock)
```

Comment:
this is one way of doing it.

another would be to look in the N+1 for the miniblocks with processing type Scheduled and construction type Final, these already have the invalid txs filtered out. 

Both should work, but the second option would simplify a bit the code here.